src/io/parsers/GROParser.js

```diff
@@ -0,0 +1,230 @@
+import * as THREE from 'three';
+import _ from 'lodash';
+import Parser from './Parser';
+import chem from '../../chem';
+import GROReader from './GROReader';
+
+
+const {
+  Complex,
+  Element,
+  Molecule,
+} = chem;
+
+/**
+ * Gromos87 file format parser.
+ *
+ * @param {Number} time           - Time in ps (optional).
+ * @param {Number} numAtoms       - Number of atoms in complex.
+ * @param {Number} residueNumber  - Number of exact residue.
+ * @param {String} residueName    - Scientific name of exact residue.
+ * @param {String} atomName       - Scientific name of exact atom.
+ * @param {Number} atomNumber     - Sorted number of exact atom.
+ * @param {Array} atomPosition    - Array which contains x, y, z position of exact atom.
+ * @param {Array} atomVelocity    - Array which contains x, y, z velocity of exact atom (optional).
+ *
+ * @param {Complex} complex       - Complex structure for unified molecule representation.
+ *
+ * @param {Molecule} molecules    - Molecules array.
+ * @param {Molecule} molecule     - Single molecule.
+ * @param {String} filetype       - Extension of data file.
+ *
+ * @exports GROParser
+ * @constructor
+ */
+class GROParser extends Parser {
+  constructor(data, options) {
+    super(data, options);
+
+    this._time = null;
+    this._numAtoms = null;
+    this._residueNumber = null;
+    this._residueName = '';
+    this._atomName = '';
+    this._atomNumber = null;
+    this._atomPosition = [];
+    this._atomVelocity = [];
+
+    this._complex = null;
+
+    this._molecules = [];
+    this._molecule = null;
+    this._options.filetype = 'gro';
+  }
+
+  /**
+   * General check for possibility of parsing.
+   * @returns {boolean} true if this file is in ascii, false otherwise
+   */
+  canProbablyParse() {
+    return _.isString(this._data);
+  }
+
+  /**
+   * Parsing title of molecule complex.
+   * NOTE: that names are ESTIMATES, there is no strict rules in Gromos87 standard for first line in input file.
+   * @param {GROReader} line - Line containing title and time.
+   */
+  _parseTitle(line) {
+    const { metadata } = this._complex;
+    metadata.id = line.readLine().trim();
+    metadata.format = 'gro';
+  }
+
+  /**
+   * Parsing line containing number of atoms information.
+   * @param {GROReader} line - Line containing number of atoms.
+   */
+  _parseNumberOfAtoms(line) {
+    this._numAtoms = line.readInt(0, line.getNext());
+    if (Number.isNaN(this._numAtoms)) {
+      this._complex.error = {
+        message: `"${line.readLine()}" is not representing atom number. Consider checking input file`,
```

Comment:
Is there a reason not to throw an exception here? Do we really need to show such a detailed info (like `readLine()`) to a user instead of a line number?

Reply:
Yes, we should abort parsing procedure at that point, thanks!